src/mappings/nabla/creation.ts

```diff
@@ -218,3 +234,121 @@ export async function createSwapFee(
     await ctx.store.save(swapFee)
     return swapFee
 }
+
+export async function createNablaSwap(
+    ctx: EventHandlerContext,
+    blockNumber: number,
+    extrinsicIndex: number | undefined,
+    timestamp: number | undefined,
+    sender: string,
+    amountIn: bigint,
+    amountOut: bigint,
+    tokenIn: string,
+    tokenOut: string,
+    to: string,
+    swapFee: NablaSwapFee | undefined
+): Promise<NablaSwap> {
+    const nablaSwapId = `${blockNumber}-${extrinsicIndex ?? ''}`
+    const nablaSwap = new NablaSwap({
+        id: nablaSwapId,
+        timestamp: new Date(timestamp ?? 0),
+        sender,
+        amountIn,
+        amountOut,
+        tokenIn,
+        tokenOut,
+        to,
+        swapFee,
+    })
+
+    await ctx.store.save(nablaSwap)
+    return nablaSwap
+}
+
+export async function createNablaBackstopLiquidityDeposit(
+    ctx: EventHandlerContext,
+    blockNumber: number,
+    extrinsicIndex: number | undefined,
+    timestamp: number | undefined,
+    sender: string,
+    poolSharesMinted: bigint,
+    amountPrincipleDeposited: bigint
+): Promise<NablaBackstopLiquidityDeposit> {
+    const backstopDepositId = `${blockNumber}-${extrinsicIndex ?? ''}`
```

Comment:
I'm not a big fan of the ID derivation in general. How about we always make it a combination of the related item (in this example, the backstop pool) + e.g. `ctx.event.id`or similar. Then we also have to pass two arguments less into each of these calls.

Reply:
A possible argument could be, e.g., [this one](https://github.com/pendulum-chain/pendulum-squids/pull/62/files#diff-0f12e301645d6153e996e498d2c74df4bc64af81718df3ab0f9d46a4e814b0daR95): here we need to request some entry from the `NablaSwapFee` table that has been created in another event shortly before – the function `swapExactTokensForTokens` of the router contract will generate the following actions:
1. the swap out pool emits the `ChargedSwapFees` event, which will be handled [in the indexer](https://github.com/pendulum-chain/pendulum-squids/pull/62/files#diff-0f7e42b3ffe1b4451fc54f2546a359c89a0d16cae8347478995936b49e50fb6eR45) by creating a `NablaSwapFee` entry
1. the router emits a `Swap` event, the handler of which is the one I link to above

Using this id, it is easy to look up the associated `NablaSwapFee`.

A counter argument is that this would not work anymore for batch transactions.

Reply:
@ebma I extracted the id generation logic to a function, but I think we should create an extra ticket for [this](https://github.com/pendulum-chain/pendulum-squids/pull/62#discussion_r1611963997) since we have to take into account batch transactions as Torsten mentioned. WDYT? @ebma @TorstenStueber

Reply:
It might be overkill to already think about batch transactions. In my view there is no ideal way to connect events that are generated during the same Nabla operation when they are actually batched in the same extrinsic. Or is there something that comes to your mind?

Maybe the best way is to avoid the situation all together and never try to connect events as we try to do here. Not sure yet what we can do instead.